module-system/sov-modules-api/src/tests.rs

```diff
@@ -44,3 +45,127 @@ fn test_hex_conversion() {
     let deserialized_pub_key = DefaultPrivateKey::from_hex(&hex).unwrap().pub_key();
     assert_eq!(priv_key.pub_key(), deserialized_pub_key)
 }
+
+struct ModuleA {
+    address: Address,
+}
+
+impl crate::ModuleInfo for ModuleA {
+    type Context = DefaultContext;
+
+    fn address(&self) -> &<Self::Context as crate::Spec>::Address {
+        &self.address
+    }
+
+    fn dependencies(&self) -> Vec<&<Self::Context as crate::Spec>::Address> {
+        vec![]
+    }
+}
+
+struct ModuleB {
+    address: Address,
+    module_a: ModuleA,
+}
+
+impl crate::ModuleInfo for ModuleB {
+    type Context = DefaultContext;
+
+    fn address(&self) -> &<Self::Context as crate::Spec>::Address {
+        &self.address
+    }
+
+    fn dependencies(&self) -> Vec<&<Self::Context as crate::Spec>::Address> {
+        vec![self.module_a.address()]
+    }
+}
+
+struct ModuleC {
+    address: Address,
+    module_a: ModuleA,
+    module_b: ModuleB,
+}
+
+impl crate::ModuleInfo for ModuleC {
+    type Context = DefaultContext;
+
+    fn address(&self) -> &<Self::Context as crate::Spec>::Address {
+        &self.address
+    }
+
+    fn dependencies(&self) -> Vec<&<Self::Context as crate::Spec>::Address> {
+        vec![self.module_a.address(), self.module_b.address()]
+    }
+}
+
+#[test]
+fn test_sorting_modules() {
+    let module_a_b = ModuleA {
+        address: Address::from([1; 32]),
+    };
+    let module_a_c = ModuleA {
+        address: Address::from([1; 32]),
+    };
+    let module_a_b_c = ModuleA {
+        address: Address::from([1; 32]),
+    };
+    let module_b_c = ModuleB {
+        address: Address::from([2; 32]),
+        module_a: module_a_b_c,
+    };
+
+    let module_a = ModuleA {
+        address: Address::from([1; 32]),
+    };
+    let module_b = ModuleB {
+        address: Address::from([2; 32]),
+        module_a: module_a_b,
+    };
+    let module_c = ModuleC {
+        address: Address::from([3; 32]),
+        module_a: module_a_c,
+        module_b: module_b_c,
+    };
+
+    let modules: Vec<(&dyn ModuleInfo<Context = DefaultContext>, &i32)> =
+        vec![(&module_b, &2), (&module_c, &3), (&module_a, &1)];
+
+    let sorted_modules = crate::sort_modules_by_dependencies(&modules).unwrap();
+
+    assert_eq!(sorted_modules, vec![&1, &2, &3]);
```

Comment:
Just nitpicking, but for `assert_eq` I usually use (expected, actual) order

Reply:
Hm... this is not consistent in the repository. I also find it easier to read in order, we "assert on sorted modules equal something expected"

Reply:
Yeah, I don't have strong opinion on that. But we need to unify on this so readability will be the same for everybody.